src/Concerns/Extendable.php

```diff
@@ -35,6 +40,71 @@ public static function hasExtend(string $name): bool
         return array_key_exists($name, static::$extends);
     }
 
+    /**
+     * Register a pipe to be applied before an expectation is checked.
+     */
+    public static function pipe(string $name, Closure $handler): void
+    {
+        self::$pipes[$name][] = $handler;
+    }
+
+    /**
+     * Register an interceptor that should replace an existing expectation.
+     */
+    public static function intercept(string $name, string|Closure $filter, Closure $handler): void
+    {
+        if (is_string($filter)) {
+            $filter = fn ($value, ...$arguments): bool => $value instanceof $filter;
+        }
+
+        self::pipe($name, function ($next, ...$arguments) use ($handler, $filter): void {
+            /** @phpstan-ignore-next-line  */
+            if ($filter($this->value, ...$arguments)) {
+                /** @phpstan-ignore-next-line  */
+                $handler = $handler->bindTo($this, get_class($this));
+
+                if($handler instanceof Closure){
+                    $handler(...$arguments);
+                }
+
+                return;
+            }
+
+            $next();
+        });
+    }
+
+    /**
+     * Checks if pipes are registered for a given expectation.
+     */
+    public static function hasPipes(string $name): bool
+    {
+        return array_key_exists($name, static::$pipes);
+    }
+
+    /**
+     * Gets the pipes that have been registered for a given expectation and binds them to a context and a scope.
+     *
+     * @return array<int, Closure>
+     */
+    private function pipes(string $name, object $context, string $scope): array
+    {
+        if (!self::hasPipes($name)) {
+            return [];
+        }
+
+        $pipes = [];
+        foreach (self::$pipes[$name] as $pipe) {
+            $pipe = $pipe->bindTo($context, $scope);
+
+            if($pipe instanceof Closure){
```

Comment:
Hmmm, we do the same thing here. Why don't we do something slightly different:

I think there is already a support class in Pest for closures. We could have our own version of bind to that never returns false and instead throws an exception. That would remove all of these checks.

Reply:
If we do that, this could be rewritten as 

return `array_map(fn ($pipe) => Closure::bindTo($context, $scope), self::$pipes[$name] ?? [])`

Reply:
I like the idea, will work on it soon!

Reply:
it seems that, also here, proper typehinting solves the issue, so no need for a support class: 

```php
private function pipes(string $name, object $context, string $scope): array
{
    return array_map(fn(Closure $pipe) => $pipe->bindTo($context, $scope), self::$pipes[$name] ?? []);
}
```

and we can get rid of the `hasPipes` method